### ISSUE TITLE
Several points in the Readme do not work using Docker

### ISSUE THREAD

**Issue by paucarrecardona** [MaverickPeter/MR_SLAM#14]
- The point **6.** "Run preprocess tools (in 3 terminals)" does not work 

Neither the `Tool` folder nor the file `robot_1.py` exists

- The point **7.** "Run loop detection module" does not work
```
# rosrun disco_ros main.py
/opt/ros/noetic/bin/rosrun: /home/LoopDetection/src/disco_ros/main.py: /home/client/miniconda3/envs/py3-mink/bin/python3: bad interpreter: No such file or directory
/opt/ros/noetic/bin/rosrun: line 150: /home/LoopDetection/src/disco_ros/main.py: Success
```

- The point **9.** "Run preprocess tools (in 3 terminals)" does not work

```
cd Costmap && source devel/setup.bash
roslaunch move_base move_base_server.launch
```
Produces the following error:
```
> roslaunch move_base move_base_server.launch
RLException: [move_base_server.launch] is neither a launch file in package [move_base] nor is [move_base] a launch file name
The traceback for the exception was written to the log file
```

What would be the minimal setup to be able to see the traversability costmap on RViz?

**Comment by MaverickPeter** [comment]
@paucarrecardona, Thanks for your following. I'll update the docker image and README this week.

**Comment by paucarrecardona** [comment]
Thanks @MaverickPeter !
 I'll test ASAP

**Comment by paucarrecardona** [comment]
@MaverickPeter 
I pulled the latest version and I tried the following:
1- "Quick Demo" from the Readme.md did not show any data in RViz, just an empty grid
2- "Full Usage" playing bag `loop-22.bag`, and only using `robot_2` scripts.
This also did not show anything beyond an empty grid in RVIz:
![Screenshot from 2023-11-13 17-03-05](https://github.com/MaverickPeter/MR_SLAM/assets/149704543/dcdcffa9-bdce-4cd4-8758-45300b520126)

I cannot see any TF being published either.

Can you help me with that?

**Comment by MaverickPeter** [comment]
@paucarrecardona 
Based on the topic list you provided, I guess the global manager and costmap converter are not working. You may need to check these modules. Also, loop-22.bag is corresponding to the robot_1 scripts.
As for costmap visualization, you need to add a map topic and choose the global_costmap topic.

**Comment by paucarrecardona** [comment]
@MaverickPeter 
Thanks for your help. 
I round this method that created the subscriptions: https://github.com/MaverickPeter/MR_SLAM/blob/main/Mapping/src/elevation_mapping_periodical/elevation_mapping/src/ElevationMapping.cpp#L100

It seems it is never called and thus the subscriptions are not created. Shall this method and all their callbacks be removed from the code or is the method supposed to be called?

**Comment by MaverickPeter** [comment]
@paucarrecardona 
In the quick demo, the elevation mapping is not called because the elevation map is preprocessed in the provided bags. You can choose /global_manager/merged_map to visualize the global elevation map.
As for full usage, the elevation mapping should be called to process the incoming lidar points.

**Comment by paucarrecardona** [comment]
@MaverickPeter 
Thanks
I'm trying to figure out why I can't generate elevation maps.
I stumbled upon something I don't understand and I was wondering if you'd be able to help me to understand the code.
As far as I understand, [void ElevationMapping::processpoints(pcl::PointCloud<Anypoint>::ConstPtr pointCloud)](
https://github.com/MaverickPeter/MR_SLAM/blob/a46cbadac6400f4a815dcabe806e02ed67d17a0f/Mapping/src/elevation_mapping_periodical/elevation_mapping/src/ElevationMapping.cpp#L247) is the main method to compute the elevation map.
As far as I understand, the height shall be stored in float [point_height[point_num];](https://github.com/MaverickPeter/MR_SLAM/blob/a46cbadac6400f4a815dcabe806e02ed67d17a0f/Mapping/src/elevation_mapping_periodical/elevation_mapping/src/ElevationMapping.cpp#L254C3-L254C33)

But after the method is finished, I can't see how the height/elevation is used. As far as I can see in the code, the height is lost and never used.

Would you be able to help me here?

**Comment by MaverickPeter** [comment]
@paucarrecardona 
You're right about the generation process. After map_.show(), the elevation and traversability are transferred into map_ defined in ElevationMap.cpp which is further visualized using visualPointMap() function. This elevation_mapping is modified from [GEM](https://github.com/ZJU-Robotics-Lab/GEM)
In my experience, this might be some configuration problem rather than a code issue,

**Comment by qiyu-lu** [comment]
@paucarrecardona  @MaverickPeter 
Hello, sorry to bother you. I encountered a similar issue when using Docker for the "Quick Demo". In RViz, only the grid lines are displayed. How can I resolve this?
